pkg/apis/core/validation/shoot.go

```diff
@@ -747,8 +747,22 @@ func ValidateKubernetesVersionUpdate(new, old string, skipMinorVersionAllowed bo
 		return allErrs
 	}
 
+	// Fill missing defaulted Kubernetes version parts from the old version, as they are only added afterwards via the mutating admission webhook.
```

Comment:
Is this true? 🤔 

I needed to build an order of execution while working on https://github.com/gardener/gardener/issues/2158. For creation, I came up with (steps like defaulting and others are omitted):
1. In-tree mutating admission plugins
2. Mutating webhooks (via the `MutatingAdmissionWebhook` admission plugin)
3. `strategy.PrepareForCreate`
4. `strategy.Validate` (storage layer validation)
5. In-tree validating admission plugins
6. Validating webhooks (via the `ValidatingAdmissionWebhook` admission plugin)

It should not be the case `strategy.Validate` / `strategy.ValidateUpdate` is invoked before mutating webhooks.

But correct me if I am missing something. 😅

Reply:
It seems that indeed your recent PRs remediated the bug and it's working now as intended. 👍